glyph/src/ttfp.rs

```diff
@@ -19,23 +19,51 @@ impl From<GlyphId> for ttfp::GlyphId {
 
 /// A pre-rendered image of a glyph, usually used for emojis or other glyphs
 /// that can't be represented only using an outline.
+#[deprecated(since = "0.2.21", note = "Deprecated in favor of `GlyphImage2`")]
 #[derive(Debug, Clone)]
 pub struct GlyphImage<'a> {
     /// Offset of the image from the normal origin (top at the baseline plus
     /// ascent), measured in pixels at the image's current scale.
     pub origin: Point,
     /// Current scale of the image in pixels per em.
     pub scale: f32,
-    /// Raw image data (not a bitmap).
+    /// Raw image data, not a bitmap in the case of [`GlyphImageFormat::Png`] format.
     pub data: &'a [u8],
     /// Format of the raw data.
     pub format: GlyphImageFormat,
 }
 
+#[allow(deprecated)]
 impl<'a> From<ttfp::RasterGlyphImage<'a>> for GlyphImage<'a> {
     fn from(img: ttfp::RasterGlyphImage<'a>) -> Self {
+        let img: GlyphImage2 = img.into();
         GlyphImage {
+            origin: img.origin,
+            scale: img.scale,
+            data: img.data,
+            format: img.format,
+        }
+    }
+}
+/// A pre-rendered image of a glyph, usually used for emojis or other glyphs
+/// that can't be represented only using an outline.
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+pub struct GlyphImage2<'a> {
+    pub origin: Point,
+    pub width: usize,
+    pub height: usize,
```

Comment:
I thought it could be convenient for caller to calculate data indices with `usize`.
```
let id = x + y * width
```

But this indeed coerced `u16` to `uszie`, should be fine to just use u16.